capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java

```diff
@@ -128,4 +135,52 @@ public static void createUser(int userId, String userName, String email) throws
     }
     return itemTypes;
   }
+
+  /*
+   * Provided a list of ItemTypes, this method will return a list of Stores
+   * with every permutation of possible items from any given store that 
+   * satisfies this list of ItemTypes.
+   *
+   * @param itemTypes list of Strings that describes the desired Item Types
+   * @return a list of all potential stores and combinations that satisfy the Item Types
+   *
+   */
+  public static List<Store> getStoresWithItems(List<String> itemTypes) {
+    List<Store> stores = new ArrayList<Store>();
+    DatabaseClient dbClient = initClient();
+    String query = "";
+    boolean first = true;
+    for (String itemType : itemTypes) {
+      List<Store> newStores = new ArrayList<Store>();
+      query = "SELECT a.ItemId, a.ItemNameAndBrand, b.StoreId, b.Price, c.Address, c.StoreName " +
+        "FROM Items a JOIN Inventory b ON a.ItemId = b.ItemId JOIN Stores c ON b.StoreId = c.StoreId " +
+        "WHERE b.ItemAvailability = 'AVAILABLE' AND a.ItemType = @itemType";
+      Statement allStatement = Statement.newBuilder(query).bind("itemType").to(itemType).build();
+      try(ResultSet allInfo = dbClient.singleUse().executeQuery(allStatement)) {
+        while(allInfo.next()) {
+          long itemId = allInfo.getLong(ITEM_ID);
+          String itemName = allInfo.getString(ITEM_NAME_AND_BRAND);
+          long storeId = allInfo.getLong(STORE_ID);
+          double itemPrice = allInfo.getDouble(PRICE);
+          String storeAddress = allInfo.getString(ADDRESS);
+          String storeName = allInfo.getString(STORE_NAME);
+          if(first) {
```

Comment:
this variable `first` is not being set to false anywhere? So the "else" part of this loop will never run. Do you need this?